app/src/main/java/com/lumisdinos/gameoffifteen/domain/repos/GameRepository.kt

```diff
@@ -0,0 +1,24 @@
+package com.lumisdinos.gameoffifteen.domain.repos
+
+import com.lumisdinos.gameoffifteen.domain.model.GameModel
+
+interface GameRepository {
+
+    suspend fun getAllGames(): List<GameModel>
+
+    suspend fun getGame(gameId: Int): GameModel?
+
+    suspend fun getGame(gameName: String): GameModel?
```

Comment:
as far as i understand your model will never be null. you don't need to use elvis

Reply:
No, it may be null if I pass wrong name